pkg/controller/cincinnati/new.go

```diff
@@ -301,6 +344,27 @@ func (k *kubeResources) newDeployment(instance *cv1alpha1.Cincinnati) *appsv1.De
 			},
 		},
 	}
+	if k.UseInitContainer(instance) {
```

Comment:
If you update `k.graphDataInitContainer` to return `nil` when the image isn't specified, then you should be able to include it with the deployment definition. Ie:

```go
				Spec: corev1.PodSpec{
...
					Containers: []corev1.Container{
						*k.graphBuilderContainer,
						*k.policyEngineContainer,
					},
					InitContainers: []corev1.Container{
						*k.graphBuilderContainer,
					},
				},
```

Reply:
Wouldn't that cause a nil pointer dereference?

Reply:
downside of not being able to make :coffee: I guess.

Reply:
Still think we can get around the `UseInitContainer` by letting `graphBuilderContainer` return `nil` and saying:

```go
if k.graphBuilderContainer != nil {
    ...Spec.InitContainers = []corev1.Container{*k.graphBuilderContainer}
}
```

Reply:
Another option would be to just have `k.initContainers = newInitContainers(instance)`, and let it return a slice of whatever length it thinks is appropriate. Then in the Deployment Spec you can always assign the value, and if it's an empty slice, no harm. Just another option, and I don't personally have a preference.